munemo/munemo.go

```diff
@@ -44,19 +54,63 @@ func newMunemo() *munemo {
 			"ya", "yi", "yu", "ye", "yo",
 			"za", "zi", "zu", "ze", "zo",
 		},
+		sign:           1,
+		symbol_values:  make(map[string]int),
 		negativeSymbol: "xa",
 		buffer:         new(bytes.Buffer),
 	}
+	for k, v := range m.symbols {
+		m.symbol_values[v] = k
+	}
+
+	return m
 }
 
 func (m *munemo) string() string {
 	return m.buffer.String()
 }
 
+func (m *munemo) int() int {
+	return m.number
+}
+
+func (m *munemo) decode(s string) {
+	arr := []byte(s)
```

Comment:
No need to convert to the byte slice. All the operations below would still work on the raw string. 
Ex: `s = s[2:len(s)]`